src/main.rs

```diff
@@ -109,6 +109,43 @@ const DEFAULT_INDEX_HTML: &'static str = r#"
 </html>
 "#;
 
+const DEFAULT_UNKNOWN_UNKNOWN_INDEX_HTML: &'static str = r#"
+<!DOCTYPE html>
+<head>
+    <meta charset="utf-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=1" name="viewport" />
+    <script>
+    var importObject = {'env': {
+        'js_raw_eval': function(pointer, length) {
+            if (window.instance === undefined) {
+                console.error("Callbacks during main() don't work for lack of access to the WASM instance, please move your code from `main` to `main_from_js`.");
+            }
+            var as_u8 = new Uint8Array(window.instance.exports.memory.buffer, pointer, length)
+            var decoder = new TextDecoder("UTF-8");
+            var s = decoder.decode(as_u8);
+            return eval(s);
+        },
+    }};
+
+    fetch("/main.wasm").then(response =>
```

Comment:
(I haven't played with this yet myself) Does rustc always generate a file called `main.wasm`? (With the Emscripten backend an `<project-name>.wasm` was generated).

Reply:
No, it generates a <project-name>.wasm. I'm mimicking what the `cargo web start` has done with the emscripten-generated JavaScript here: serve it under a constant path no matter its name, so that a static index.html can pick it up.

The server already serves the file under its original file name as well, I can (and probably should) adapt the index.html to not be static but have the file name dynamic. (I tend to be careful when generating HTML, but with the name coming from the crate's name, we're probably safe against HTML injection ;-) )